Flaky test: TestPublishGroupMessageConcurrent

https://github.com/xmtp/xmtpd/actions/runs/13909345688/job/38919777996

```
=== RUN   TestPublishGroupMessageConcurrent
2025-03-17T20:41:07.844Z	INFO	blockchain/blockchainPublisher.go:71	Starting server with blockchain nonce: 5
2025-03-17T20:41:07.844Z	DEBUG	blockchain/nonceManager_test.go:67	Generated Nonce	{"nonce": 5}
2025-03-17T20:41:07.844Z	DEBUG	blockchain/nonceManager_test.go:67	Generated Nonce	{"nonce": 6}
2025-03-17T20:41:07.845Z	DEBUG	blockchain/nonceManager_test.go:67	Generated Nonce	{"nonce": 7}
2025-03-17T20:41:07.845Z	DEBUG	blockchain/nonceManager_test.go:67	Generated Nonce	{"nonce": 8}
2025-03-17T20:41:07.845Z	DEBUG	blockchain/nonceManager_test.go:67	Generated Nonce	{"nonce": 9}
2025-03-17T20:41:07.845Z	DEBUG	blockchain/nonceManager_test.go:67	Generated Nonce	{"nonce": 10}
2025-03-17T20:41:07.845Z	DEBUG	blockchain/nonceManager_test.go:67	Generated Nonce	{"nonce": 11}
2025-03-17T20:41:07.845Z	DEBUG	blockchain/nonceManager_test.go:67	Generated Nonce	{"nonce": 12}
2025-03-17T20:41:07.845Z	DEBUG	blockchain/nonceManager_test.go:67	Generated Nonce	{"nonce": 13}
2025-03-17T20:41:07.846Z	DEBUG	blockchain/nonceManager_test.go:67	Generated Nonce	{"nonce": 14}
2025-03-17T20:41:07.846Z	DEBUG	blockchain/nonceManager_test.go:67	Generated Nonce	{"nonce": 15}
2025-03-17T20:41:07.846Z	DEBUG	blockchain/nonceManager_test.go:67	Generated Nonce	{"nonce": 16}
2025-03-17T20:41:07.847Z	DEBUG	blockchain/nonceManager_test.go:67	Generated Nonce	{"nonce": 17}....
2025-03-17T20:41:10.225Z	ERROR	GroupBlockchainPublisher	blockchain/client.go:90	waiting for transaction	{"contractAddress": "0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512", "hash": "0x51b9fac962bcadbb985173a9bb3ef9f739cbda58ed33e911d6dfb123a681e730"}
github.com/xmtp/xmtpd/pkg/blockchain.WaitForTransaction
	/home/runner/work/xmtpd/xmtpd/pkg/blockchain/client.go:90
github.com/xmtp/xmtpd/pkg/blockchain.(*BlockchainPublisher).PublishGroupMessage.func2
	/home/runner/work/xmtpd/xmtpd/pkg/blockchain/blockchainPublisher.go:140
github.com/xmtp/xmtpd/pkg/blockchain.withNonce[...]
	/home/runner/work/xmtpd/xmtpd/pkg/blockchain/blockchainPublisher.go:298
github.com/xmtp/xmtpd/pkg/blockchain.(*BlockchainPublisher).PublishGroupMessage
	/home/runner/work/xmtpd/xmtpd/pkg/blockchain/blockchainPublisher.go:127
github.com/xmtp/xmtpd/pkg/blockchain_test.TestPublishGroupMessageConcurrent.func1
	/home/runner/work/xmtpd/xmtpd/pkg/blockchain/blockchainPublisher_test.go:177
    blockchainPublisher_test.go:199: Errors encountered:
        timed out
--- FAIL: TestPublishGroupMessageConcurrent (2.53s)
```

--- a/dev/docker/docker-compose.yml
+++ b/dev/docker/docker-compose.yml
@@ -16,7 +16,7 @@ services:
   chain:
     platform: linux/amd64
     image: ghcr.io/foundry-rs/foundry
-    entrypoint: ["anvil", "--host", "0.0.0.0"]
+    entrypoint: ["anvil", "--host", "0.0.0.0", "--mixed-mining", "--block-time", "1"]
     working_dir: /anvil
     ports:
       - 7545:8545

--- a/pkg/testutils/anvil/anvil.go
+++ b/pkg/testutils/anvil/anvil.go
@@ -48,7 +48,15 @@ func waitForAnvil(t *testing.T, url string) {
 func StartAnvil(t *testing.T, showLogs bool) (string, func()) {
 	port := networkTestUtils.FindFreePort(t)
 
-	cmd := exec.Command("anvil", "--port", fmt.Sprintf("%d", port))
+	// we need mixed mining to work around https://github.com/xmtp/xmtpd/issues/643
+	cmd := exec.Command(
+		"anvil",
+		"--port",
+		fmt.Sprintf("%d", port),
+		"--mixed-mining",
+		"--block-time",
+		"1",
+	)
 	if showLogs {
 		// (Optional) You can capture stdout/stderr for logs:
 		cmd.Stdout = os.Stdout

